schema_bump_dry_run_scripts/genes/gene_bump_dry_run.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+import logging
+import os
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple
+
+import tiledb as tiledb
+from jinja2 import Template
+
+from cellxgene_schema_cli.cellxgene_schema.env import ONTOLOGY_DIR
+from cellxgene_schema_cli.cellxgene_schema.ontology import SupportedOrganisms
+from schema_bump_dry_run_scripts.common import (
+    BASE_API,
+    fetch_private_collections,
+    fetch_private_dataset,
+    fetch_public_datasets,
+    get_headers,
+)
+from schema_bump_dry_run_scripts.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger()
+
+ctx = tiledb.default_ctx({"vfs.s3.region": "us-west-2"})
+
+
+def generate_report(data) -> str:
+    file_path = os.path.dirname(os.path.realpath(__file__))
+    with open(os.path.join(file_path, "report_template.j2"), "r") as fp:
+        report = fp.read()
+
+    j2_template = Template(report, trim_blocks=True, lstrip_blocks=True)
+    report = j2_template.render(data)
+    return report
+
+
+def get_genes(dataset: dict) -> List[str]:
+    """
+    Uses tiledb to get the genes for a dataset. This method is slower, but does not add a dependency on the explorer. It
+    is also free if we run computer with in the same AWS region.
+
+    :param dataset: dataset metadata
+    :param stage: prod, staging, or dev
+    :return:
+    """
+
+    if dataset.get("s3_uri"):
+        s3_path = dataset.get("s3_uri") + "/var"
+    elif dataset.get("dataset_assets"):
+        s3_path = [asset["s3_uri"] for asset in dataset["dataset_assets"] if asset["filetype"] == "CXG"][0] + "/var"
+    else:
+        dataset_version_id = dataset["dataset_version_id"]
+        stage = os.getenv("corpus_env", default="dev")
+        s3_path = f"s3://hosted-cellxgene-{stage}/{dataset_version_id}.cxg/var"
+
+    logger.debug(f"Fetching genes from {s3_path}")
+    with tiledb.open(s3_path, "r") as var:
+        var_df = var.df[:]
+        suffix = 0
+        while f"name_{suffix}" not in var_df.columns:
+            suffix += 1
+        index_name = f"name_{suffix}"
+        stored_genes = var_df[index_name].to_list()
+    return stored_genes
+
+
+def get_diff_map() -> Dict[str, List[str]]:
+    # list all of the files ending with diff.txt in the cellxgene_schema/ontology_files directory
+    # for each file, open it and read the contents into a dictionary
+    diff_map = {}
+    suffix = "_diff.txt"
+    files = os.listdir(ONTOLOGY_DIR)
+    for file in files:
+        if file.endswith(suffix):
+            with open(f"{ONTOLOGY_DIR}/{file}") as fp:
+                organism = getattr(SupportedOrganisms, file.removesuffix(suffix).upper()).value
```

Comment:
This is the comment I'm referring to--the push tests are failing because GHA is set to python 3.8, but this function was introduced in python 3.9